packages/subgraphs/quest-subgraph/src/mapping.ts

```diff
@@ -33,35 +69,54 @@ export function handleQuestCreated(event: QuestCreated): void {
     }
   }
 
-  //   let collateral = metadata.get("collateral");
-  //   questEntity.questMetaCollateralPercentage = collateral
-  //     ? collateral.toBigInt()
-  //     : null;
+  questEntity.save();
+}
+
+export function handleQuestCreatedV3(event: QuestCreatedV3): void {
+  let questEntity = new QuestEntity(event.params.questAddress.toHex());
+
+  questEntity.questAddress = event.params.questAddress.toHexString();
+  questEntity.questTitle = event.params.questTitle;
+  questEntity.questDetailsRef = event.params.questDetailsRef;
+  questEntity.questRewardTokenAddress = event.params.rewardTokenAddress;
+  questEntity.questExpireTimeSec = event.params.expireTime;
+  questEntity.creationTimestamp = event.params.creationTime;
+  questEntity.depositToken = event.params.depositToken;
+  questEntity.depositAmount = event.params.depositAmount;
 
-  //   let tags = metadata.get("tags");
-  //   questEntity.questMetaTags = tags
-  //     ? tags
-  //         .toArray()
-  //         .filter((x) => !x.isNull())
-  //         .map<string>((x) => x.toString())
-  //     : [];
+  if (!event.params.questDetailsRef) {
+    questEntity.questDescription = "";
+  } else {
+    // Fetching quest description with IPFS
+    let questDataBytes: Bytes | null = null;
+    let tryCount = 0;
+    while (!questDataBytes && tryCount < 3) {
+      // 3 tries in total (180 sec for each try)
+      questDataBytes = ipfs.cat(event.params.questDetailsRef.toString());
+      tryCount = tryCount + 1;
+    }
+    if (questDataBytes) {
+      let description = questDataBytes.toString();
+      questEntity.questDescription = description ? description.toString() : "";
+    } else {
+      // Continue with empty description
+      questEntity.questDescription = "";
+    }
+  }
 
   questEntity.save();
+}
 
-  // Note: If a handler doesn't require existing field values, it is faster
-  // _not_ to load the entity from the store. Instead, create it fresh with
-  // `new Entity(...)`, set the fields that should be updated and save the
-  // entity back to the store. Fields that were not set or unset remain
-  // unchanged, allowing for partial updates to be applied.
+export function handleDepositChanged(event: DepositChanged): void {
+  let questEntity = new DepositEntity(
+    event.params.timestamp.toString() +
+      event.params.token.toHex() +
+      event.params.amount.toHex()
```

Comment:
If that is the ID, what you think use template string `${}_${}_${}` spliting with underline or hifen in case if we need split it could be easy